### ISSUE TITLE
is_complex_layout/pages_with_columns reported but multi-column pages still flattened into one interleaved stream (bogus table)

### ISSUE THREAD

**Issue by yonikremer** [firecrawl/pdf-inspector#498]
# is_complex_layout/pages_with_columns are reported, but multi-column pages are still flattened into one interleaved stream (wrong reading order, wrapped in a bogus table)

**pdf-inspector 1.17.0** (PyPI wheel, Windows x64, CPython 3.13).

A single-page, text-based PDF whose layout is **two body-text columns plus a band of three independent stat columns** is detected as complex, but the extracted markdown interleaves the columns line-by-line into one reading stream and wraps the whole page in a spurious GFM `<table>` — the stat numbers are detached from their captions and the two text columns are merged.

## Repro

Example PDF: https://drive.google.com/file/d/1oCsKkip9762JOT6XFcwtlZG1q3hRCZBi/view (single page; this is the same file used in #497).

```python
import pdf_inspector
r = pdf_inspector.process_pdf("hebrew_letter_splitting_repro.pdf")
print(r.is_complex_layout, r.pages_with_columns, r.pages_with_tables)
# True  [1]  [1]
print(r.markdown)
```

Actual markdown (excerpts; whole page is emitted as one `<table>`):

```text
|י ר | הה י ע ר כ ו ת התקש ו |…| … |
| … | … 48% … | … כ - 4% … | … כ - 40,000 | …
```

## What the page actually is (word coordinates, PyMuPDF, same page)

The page has real column structure — multiple independent x-columns carry text at the **same y**:

```text
Two body-text columns (both at y≈468):
  right column:  'מועסקים'(x≈416) 'במשרדי'(x≈385)   …
  left  column:  'מכלל'(x≈207) 'עובדי'(x≈185) 'משרדי'(x≈158) …

Stat band (three independent columns at y≈439, each with its own caption BELOW it):
  x≈185 → '48%'
  x≈301 → '4%'
  x≈403 → '40,000'
```

Expected reading order: complete each column top→bottom (and each number with the caption beneath it), then advance to the next column.

Actual: pdf-inspector interleaves column A and column B line by line, and flattens the three stat columns plus their captions into one mis-associated table row — numbers and labels that belong to different visual columns end up as neighbouring cells, and the two prose columns are merged into one continuous paragraph.

## For reviewers who don't read Hebrew

Ignore the script — the coordinates above are enough: at a single `y`, there are words at x≈158–310 **and** at x≈385–451 with a clear gutter between them (two columns), and the stat band has three numbers at three distinct x positions each with a caption below. Any column-aware extractor keeps those grouped; pdf-inspector merges them into one stream. (The `48% / 4% / 40,000 / 57%−51%` figures are LTR numbers, so the mis-grouping is visible without reading any Hebrew.)

## Relationship to #497

Distinct defect. #497 is word-internal fragmentation (spaces inserted *inside* words). This one is block/column-level ordering: even text that is otherwise intact is emitted in the wrong sequence because column boundaries are ignored — despite `is_complex_layout`/`pages_with_columns` being set.
